src/Components/3DV/SceneView.tsx

```diff
@@ -1680,6 +1683,132 @@ function SceneView(props: ISceneViewProps, ref) {
+                                    previouslyTransformedElement.transform.rotation.z;
+                                // setting position back to original
+                                prevTransParentMesh.position.x =
+                                    previouslyTransformedElement.transform.position.x;
+                                prevTransParentMesh.position.y =
+                                    previouslyTransformedElement.transform.position.y;
+                                prevTransParentMesh.position.z =
+                                    previouslyTransformedElement.transform.position.z;
+                            }
+
+                            // set up to remove from previouslyTransformedElements
+                            previouslyTransformedElement.meshId = null;
+                        }
+                    }
+                );
+                // remove all elements with parent mesh id of null (aka was already reset)
+                previouslyTransformedElements.current = previouslyTransformedElements.current.filter(
+                    (cPTE) => cPTE.meshId != null
+                );
+
+                transformedElementItems.forEach((transformedElementItem) => {
+                    const meshIds = transformedElementItem.meshIds;
+                    const parentMeshId = transformedElementItem.parentMeshId;
+                    meshIds.forEach((meshId) => {
+                        if (meshId != parentMeshId) {
+                            // set parent of each mesh (that isn't the designated parent) to parent mesh
+                            meshMap.current?.[meshId].setParent(
+                                meshMap.current?.[parentMeshId]
+                            );
+                        }
+                    });
+                    transformMesh(transformedElementItem); // only call transform on parent mesh
+                });
+            } catch {
+                console.warn('unable to transform mesh');
+            }
+            if (debugLogging) {
+                console.timeEnd('transforming meshes');
+            }
+        }
+    }, [transformedElementItems, isLoading]);
+
+    const transformMesh = (transformedElementItem: TransformedElementItem) => {
+        const parentMesh: BABYLON.Mesh =
+            meshMap.current?.[transformedElementItem.parentMeshId];
+        if (!parentMesh) {
+            return;
+        }
+
+        parentMesh.rotationQuaternion = null; // need to do this to change mesh.rotation directly
+
+        const pTParentMeshIds: string[] = previouslyTransformedElements.current.map(
+            (pTE) => pTE.meshId
+        );
+
+        const transform = transformedElementItem.transform;
+
+        // only add parentMesh to previouslyTransformedElements ONCE for the ORIGINAL status
+        if (!pTParentMeshIds.includes(transformedElementItem.parentMeshId)) {
+            previouslyTransformedElements.current.push({
```

Comment:
could pull this creation logic to the util as well. Pass in a trasnform/mesh and return back the object?

Reply:
done! (also used them to refactor gizmo functions)